tests/reduction/reduction_with_several_reductions_in_kernel.h

```diff
@@ -0,0 +1,637 @@
+                        {sycl::property::reduction::initialize_to_identity()})};
+    auto reduction_with_ptr_to_variable_1{
+        sycl::reduction(ptr_for_variable_1.get(), functor_3())};
+    auto reduction_with_ptr_to_variable_2{sycl::reduction(
+        ptr_for_variable_2.get(), reduction_common::identity_value, functor_4(),
+        {sycl::property::reduction::initialize_to_identity()})};
+    auto buf_accessor =
+        initial_buf.template get_access<sycl::access_mode::read>(cgh);
+    auto lambda{get_lambda_for_6_reductions<RangeT>(buf_accessor)};
+    cgh.parallel_for<kernel<test_case>>(
+        range, reduction_with_buffer_1, reduction_with_buffer_2,
+        reduction_with_span_1, reduction_with_span_2,
+        reduction_with_ptr_to_variable_1, reduction_with_ptr_to_variable_2,
+        lambda);
+  });
+  check_output_value(output_buffer_1.get_host_access()[0],
+                     expected_value_for_buffer_1, __LINE__, test_case, log);
+  check_output_value(output_buffer_2.get_host_access()[0],
+                     expected_value_for_buffer_2, __LINE__, test_case, log);
+  check_output_value(*ptr_for_variable_1.get(),
+                     expected_value_for_ptr_to_variable_1, __LINE__, test_case,
+                     log);
+  check_output_value(*ptr_for_variable_2.get(),
+                     expected_value_for_ptr_to_variable_2, __LINE__, test_case,
+                     log);
+  for (size_t i = 0; i < number_elements; i++) {
+    check_output_value(mem_for_span_1.get()[i], expected_value_for_span_1,
+                       __LINE__, test_case, log);
+    check_output_value(mem_for_span_2.get()[i], expected_value_for_span_2,
+                       __LINE__, test_case, log);
+  }
+}
+
+/** @brief Run tests from two to six reducers with chosed range type
+ *  @tparam RangeT Type range
+ *  @param range sycl::range or sycl::nd_range class object
+ *  @param queue sycl::queue class object
+ *  @param log sycl_cts::util::logger class object
+ */
+template <typename RangeT>
+void run_all_tests_for_chosen_range(RangeT range, sycl::queue& queue,
+                                    sycl_cts::util::logger& log) {
+  run_test_for_two_reductions(range, queue, log);
+  run_test_for_three_reductions(range, queue, log);
+  run_test_for_four_reductions(range, queue, log);
+  run_test_for_five_reductions(range, queue, log);
+  run_test_for_six_reductions(range, queue, log);
+}
+
+/** @brief Run tests run all test with sycl::range and sycl::nd_range types
+ *  @param queue sycl::queue class object
+ *  @param log sycl_cts::util::logger class object
+ */
+void run_all_tests(sycl::queue& queue, sycl_cts::util::logger& log) {
+  run_all_tests_for_chosen_range(reduction_common::range, queue, log);
+  run_all_tests_for_chosen_range(reduction_common::nd_range, queue, log);
+}
+
+}  // namespace reduction_with_several_reductions_in_kernel_h
+
+#endif  // __SYCL_CTS_TEST_REDUCTION_WITH_ALL_FUNCTORS_H
```

Comment:
All this file should be factorized a lot by writing generic code instead.

Reply:
Could You clarify how we can factorized this file?

Reply:
Something on top of what is in this private discussion when we were polishing the API: https://gitlab.khronos.org/sycl/Specification/-/issues/452#note_288507

But as said before, we can keep the current version.
Of course, if you are very motivated by doing a metaprogramming exercise, I will not stop you. :-)